drivers/include/periph/qspi.h

```diff
@@ -0,0 +1,267 @@
+    SFLASH_CMD_WRITE_DISABLE = 0x04,        /**< Revoke Write Enable                 */
+
+    SFLASH_CMD_ERASE_SECTOR = 0x20,         /**< sector (4k) erase                   */
+    SFLASH_CMD_ERASE_BLOCK = 0xD8,          /**< block (64k) erase                   */
+    SFLASH_CMD_ERASE_CHIP = 0xC7,           /**< whole chip erase                    */
+
+    SFLASH_CMD_4_BYTE_ADDR = 0xB7,          /**< enable 32 bit addressing */
+    SFLASH_CMD_3_BYTE_ADDR = 0xE9,          /**< enable 24 bit addressing */
+} spi_flash_cmd_t;
+
+/**
+ * @brief   Default SPI device access macro
+ */
+#ifndef QSPI_DEV
+#define QSPI_DEV(x)     (x)
+#endif
+
+/**
+ * @brief   Define global value for undefined QSPI device
+ */
+#ifndef QSPI_UNDEF
+#define QSPI_UNDEF      (UINT_MAX)
+#endif
+
+/**
+ * @brief   Default type for QSPI devices
+ */
+#ifndef HAVE_QSPI_T
+typedef unsigned int qspi_t;
+#endif
+
+/**
+ * @brief   Available QSPI modes, defining the configuration of clock polarity
+ *          and clock phase
+ *
+ * RIOT is using the mode numbers as commonly defined by most vendors
+ * (https://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus#Mode_numbers):
+ *
+ * - MODE_0: CPOL=0, CPHA=0 - The first data bit is sampled by the receiver on
+ *           the first SCK rising SCK edge (this mode is used most often).
+ * - MODE_1: CPOL=0, CPHA=1 - The first data bit is sampled by the receiver on
+ *           the second rising SCK edge.
+ * - MODE_2: CPOL=1, CPHA=0 - The first data bit is sampled by the receiver on
+ *           the first falling SCK edge.
+ * - MODE_3: CPOL=1, CPHA=1 - The first data bit is sampled by the receiver on
+ *           the second falling SCK edge.
+ */
+#ifndef HAVE_QSPI_MODE_T
+typedef enum {
+    QSPI_MODE_0 = 0,        /**< CPOL=0, CPHA=0 */
+    QSPI_MODE_1,            /**< CPOL=0, CPHA=1 */
+    QSPI_MODE_2,            /**< CPOL=1, CPHA=0 */
+    QSPI_MODE_3             /**< CPOL=1, CPHA=1 */
+} qspi_mode_t;
+#endif
+
+/**
+ * @brief   Standard QSPI erase block sizes
+ */
+#ifndef HAVE_QSPI_ERASE_SIZE_T
+typedef enum {
+    QSPI_ERASE_4K   = SFLASH_CMD_ERASE_SECTOR,
+    QSPI_ERASE_64K  = SFLASH_CMD_ERASE_BLOCK,
+    QSPI_ERASE_CHIP = SFLASH_CMD_ERASE_CHIP,
+} qspi_erase_size_t;
+#endif
+
+/**
+ * @brief   Configuration Options for QSPI
+ * @{
+ */
+#define QSPI_FLAG_1BIT          (0x0)   /**< single data line           */
+#define QSPI_FLAG_2BIT          (0x1)   /**< two parallel data lines    */
+#define QSPI_FLAG_4BIT          (0x2)   /**< four parallel data lines   */
+#define QSPI_FLAG_8BIT          (0x3)   /**< eight parallel data lines  */
+
+#define QSPI_FLAG_BIT_MASK      (0x3)   /**< mask to get data lines from flags */
+#define QSPI_FLAG_BIT(flags)    ((flags) & QSPI_FLAG_BIT_MASK)
+
+#define QSPI_FLAG_DDR           (0x4)   /**< use Double Data Rate mode  */
+
+#define QSPI_FLAG_ADDR_32BIT    (0x8)   /**< use 32 bit addressing      */
```

Comment:
This also needs a flag to signal that the address can also be spread over the data lines. In both the nrf52 and the stm32 peripheral this is a separate configuration setting and behaviour of various memory chips are slightly different when the address is also spread over the available data lines.

```suggestion
#define QSPI_FLAG_ADDR_32BIT    (0x8)   /**< use 32 bit addressing      */
#define QSPI_FLAG_ADDR_MULTILINE  (0x10) /**< Use parallel data lines for the address */
```